### PR TITLE
Feature parity with permissions-graph for data-permissions table

### LINKED ISSUE #37604: Reach feature parity with data permissions graph when reading from `data_permissions`
```[tasklist]
### Tasks
- [x] run the api off of the new perm graph, including GET by group or db id
- [x] Fix issue where db is block, and we update a table on it. we should not set block on the other tables.
```

### PR DESCRIPTION
**Pull request by escherize**
resolves #37604

This PR reads a data-permissions style permission graph, and transforms it into an API-style permission graph, that involves a number of steps, as can be seen [here](https://github.com/metabase/metabase/pull/37837/files#diff-40bbdb0dde08853ee3dff65b5c2316c19c9474f31833ad4834e918481d39b02aR146-R153).

Most information still comes back from data-perms/permission-graph but admin info, sandbox info, and impersonation info need to be queried for separately.

- [x] Sanity check that the UI cannot be broken
- [x] Match the pre-existing [Malli schema](https://github.com/metabase/metabase/blob/01-17-run_permissions_graph_from_values_in_the_data-permissions_table/src/metabase/api/permission_graph.clj#L132-L136) for the api perm graph
- [x] Move `audit-db-id` into config namespace. This fixes a cyclic dependency, and now it's next to the internal metabase user id, too. Also good to move since we will eventually remove the perms namespace, so the fewer things in there now the better.
- [x] Ability to return api perm graph by-db and by-group-id.
    - Tests expect their shape of the api perm graph to have (and to _not_
have) certain db and group ids.
    - `add-sandboxes-to-permissions-graph`, `add-impersonations-to-permissions-graph`, and `add-admin-perms-to-permissions-graph` weren't respecting
the options, so we need to pass them in and filter _their_ queries too.
This fixes a number of tests.
- [x] Adds admin perms, (these are no longer stored in the db, nor do they come in on the graph) as many tests require it.
- [x] When updating a data-access perm on table in a blocked db, set the other tables in the db to `:no-self-service` (not `:block`)
- [ ] Update failing tests (doing)
    - Many tests expect to be able to insert paths and recieve a new api perm-graph

### CODE REVIEW

**Comment by escherize** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/metabase/metabase/37837?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `master`
  * **PR #37473** <a href="https://app.graphite.dev/github/pr/metabase/metabase/37473?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #37837** <a href="https://app.graphite.dev/github/pr/metabase/metabase/37837?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
    * **PR #37671** <a href="https://app.graphite.dev/github/pr/metabase/metabase/37671?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #37549** <a href="https://app.graphite.dev/github/pr/metabase/metabase/37549?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #37426** <a href="https://app.graphite.dev/github/pr/metabase/metabase/37426?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #37427** <a href="https://app.graphite.dev/github/pr/metabase/metabase/37427?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #37379** <a href="https://app.graphite.dev/github/pr/metabase/metabase/37379?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #37619** <a href="https://app.graphite.dev/github/pr/metabase/metabase/37619?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #37377** <a href="https://app.graphite.dev/github/pr/metabase/metabase/37377?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #37191** <a href="https://app.graphite.dev/github/pr/metabase/metabase/37191?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #37378** <a href="https://app.graphite.dev/github/pr/metabase/metabase/37378?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/metabase/metabase/37837?utm_source=stack-comment).

**Comment by escherize** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/metabase/metabase/37837?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `master`
  * **PR #37473** <a href="https://app.graphite.dev/github/pr/metabase/metabase/37473?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #37837** <a href="https://app.graphite.dev/github/pr/metabase/metabase/37837?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
    * **PR #37671** <a href="https://app.graphite.dev/github/pr/metabase/metabase/37671?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #37549** <a href="https://app.graphite.dev/github/pr/metabase/metabase/37549?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #37426** <a href="https://app.graphite.dev/github/pr/metabase/metabase/37426?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #37427** <a href="https://app.graphite.dev/github/pr/metabase/metabase/37427?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #37379** <a href="https://app.graphite.dev/github/pr/metabase/metabase/37379?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #37619** <a href="https://app.graphite.dev/github/pr/metabase/metabase/37619?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #37377** <a href="https://app.graphite.dev/github/pr/metabase/metabase/37377?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #37191** <a href="https://app.graphite.dev/github/pr/metabase/metabase/37191?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #37378** <a href="https://app.graphite.dev/github/pr/metabase/metabase/37378?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/metabase/metabase/37837?utm_source=stack-comment).

File: enterprise/backend/test/metabase_enterprise/advanced_permissions/models/permissions/block_permissions_test.clj [hunk 1]
```diff
@@ -33,29 +39,29 @@
...
-                (is (= (merge {:schemas :block}
-                              ;; block perms won't affect the value of `:native`; if a given group has both
-                              ;; `/db/1/` and `/block/db/1/` then the graph will come back with `:native
-                              ;; :write` and `:schemas :block`. This state isn't normally allowed, but the
-                              ;; graph code doesn't currently correct it if it happens. Not sure it's worth
-                              ;; the extra code complexity since it should never happen in the first place.
-                              (when (= path (perms/data-perms-path (mt/id)))
-                                {:native :write}))
-                       (perms group-id)))))))))))
+      (testing (str message "\n")
```
Comment (escherize) [inline enterprise/backend/test/metabase_enterprise/advanced_permissions/models/permissions/block_permissions_test.clj:42]: The only change is that this testing message gets printed now

**Comment by noahmoss** [comment]
Closing in favor of https://github.com/metabase/metabase/pull/38692

**Pull request closed without merging**